src/panoptes/pocs/focuser/astromechanics.py

```diff
@@ -0,0 +1,263 @@
+            may have a focuser adaptor attached. Default is '/dev/tty.usbserial-00*?'
+
+    Additional positonal and keyword arguments are passed to the base class, AbstractFocuser. See
+    that class' documentation for a complete list.
+    """
+
+    # Class variable to cache the device node scanning results
+    _adaptor_nodes = None
+
+    # Class variable to store the device nodes already in use. Prevents scanning known focuser adaptors and
+    # acts as a check against those ones assigned to incorrect ports.
+    _assigned_nodes = []
+
+    def __init__(self,
+                 name='Astromechanics Focuser Controller',
+                 model='astromechanics',
+                 initial_position=None,
+                 dev_node_pattern='/dev/tty.usbserial-00*',
+                 *args, **kwargs):
+        super().__init__(name=name, model=model, *args, **kwargs)
+        self.logger.debug('Initialising Astromechanics Lens Controller')
+
+        # Check that this node hasn't already been assigned to another adaptor
+        if self.port in Focuser._assigned_nodes:
+            message = 'Device node {} already in use!'.format(self.port)
+            self.logger.error(message)
+            warn(message)
+            return
+
+        try:
+            self.connect(self.port)
+        except (serial.SerialException,
+                serial.SerialTimeoutException,
+                AssertionError) as err:
+            message = 'Error connecting to {} on {}: {}'.format(self.name, self.port, err)
+            self.logger.error(message)
+            warn(message)
+            return
+
+        Focuser._assigned_nodes.append(self.port)
+        self._is_moving = False
+        self._initialise()
+
+    def __del__(self):
+        with suppress(AttributeError):
+            device_node = self.port
+            Focuser._assigned_nodes.remove(device_node)
+            self.logger.debug('Removed {} from assigned nodes list'.fomat(device_node))
+        with suppress(AttributeError):
+            self._serial_port.close()
+            self.logger.debug('Closed serial port {}'.format(self._port))
+
+    ##################################################################################################
+    # Properties
+    ##################################################################################################
+
+    @property
+    def is_connected(self):
+        """
+        Checks status of serial port to determine if connected.
+        """
+        connected = False
+        if self._serial_port:
+            connected = self._serial_port.isOpen()
+        return connected
+
+    @AbstractFocuser.position.getter
+    def position(self):
+        """
+        Returns current focus position in the lens focus encoder units.
+        """
+        response = self._send_command('P#', response_length=1)[0].replace("#", "")
+        return response
```

Comment:
The other focusers are returning ints here.

```suggestion
        return int(response)
```